packages/deepbook-wrapper/sources/fee.move

```diff
@@ -282,118 +304,92 @@ public fun estimate_full_fee_market<BaseToken, QuoteToken, ReferenceBaseAsset, R
+/// - discount_rate: Actual discount rate applied to protocol fee
 public(package) fun estimate_full_order_fee_core(
-    protocol_fee_rate: u64,
-    is_pool_whitelisted: bool,
     balance_manager_deep: u64,
     deep_in_wallet: u64,
     deep_required: u64,
     sui_per_deep: u64,
+    protocol_taker_fee_rate: u64,
+    order_amount: u64,
+    max_discount_rate: u64,
 ): (u64, u64, u64) {
-    // Determine if user needs to use wrapper DEEP reserves
-    let will_use_wrapper_deep = balance_manager_deep + deep_in_wallet < deep_required;
-
-    if (is_pool_whitelisted || !will_use_wrapper_deep) {
-        (0, 0, 0) // No fee for whitelisted pools or when user provides all DEEP
-    } else {
-        // Calculate the amount of DEEP to take from reserves
-        let deep_from_reserves = deep_required - balance_manager_deep - deep_in_wallet;
-
-        // Calculate fee based on order amount, including both protocol fee and deep reserves coverage fee
-        calculate_full_order_fee(protocol_fee_rate, sui_per_deep, deep_from_reserves)
-    }
-}
+    let deep_from_reserves = if (balance_manager_deep + deep_in_wallet < deep_required)
+        deep_required - balance_manager_deep - deep_in_wallet else 0;
 
-/// Calculates the total fee amount in SUI coins for an order using DEEP from reserves
-/// Combines both the deep reserves coverage fee and protocol fee
-///
-/// Parameters:
-/// - protocol_fee_rate: Protocol fee rate in billionths
-/// - sui_per_deep: Current DEEP/SUI price from reference pool
-/// - deep_from_reserves: Amount of DEEP taken from wrapper reserves
-///
-/// Returns:
-/// - u64: Total fee amount in SUI coins (reserves coverage fee + protocol fee)
-/// - u64: Deep reserves coverage fee
-/// - u64: Protocol fee
-///
-/// The total fee consists of:
-/// 1. Deep reserves coverage fee: SUI equivalent of borrowed DEEP
-/// 2. Protocol fee: Additional fee calculated as percentage of borrowed DEEP
-public(package) fun calculate_full_order_fee(
-    protocol_fee_rate: u64,
-    sui_per_deep: u64,
-    deep_from_reserves: u64,
-): (u64, u64, u64) {
-    // Calculate the deep reserves coverage fee
     let deep_reserves_coverage_fee = calculate_deep_reserves_coverage_order_fee(
         sui_per_deep,
         deep_from_reserves,
     );
 
-    // Calculate the protocol fee
-    let protocol_fee = calculate_protocol_fee(
-        protocol_fee_rate,
-        sui_per_deep,
+    let discount_rate = calculate_discount_rate(
+        max_discount_rate,
         deep_from_reserves,
+        deep_required,
     );
 
-    let total_fee = deep_reserves_coverage_fee + protocol_fee;
+    // Calculate protocol fee assuming order is fully taker to show fee upper limit
+    // Apply user's discount to the calculated fee
+    let (protocol_fee, _, _) = calculate_protocol_fees(
+        1_000_000_000, // 100% taker ratio
+        0, // 0% maker ratio
+        protocol_taker_fee_rate,
+        0, // no need to specify maker fee rate for 0% maker ratio
```

Comment:
Not sure we could do something here, but we probably should warn that we must not specify maker fee rate, otherwise it would lead to wrong calculations.